trestle/core/commands/author/ssp.py

```diff
@@ -0,0 +1,364 @@
+        sections_help_str = 'Comma separated list of section:alias pairs for sections to output'
+        self.add_argument('-s', '--sections', help=sections_help_str, required=False, type=str)
+        task_help_str = 'Name of the generated ssp markdown folder'
+        self.add_argument('-o', '--output', help=task_help_str, required=True, type=str)
+        verbose_help_str = 'Display verbose output'
+        self.add_argument('-v', '--verbose', help=verbose_help_str, required=False, action='count', default=0)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        log.set_log_level_from_args(args)
+        trestle_root = fs.get_trestle_project_root(pathlib.Path.cwd())
+        if not trestle_root:
+            logger.warning(f'Current working directory {pathlib.Path.cwd()} is not with a trestle project.')
+            return 1
+        if not fs.allowed_task_name(args.output):
+            logger.warning(f'{args.output} is not an allowed directory name')
+            return 1
+
+        if args.mode == 'setup':
+            profile: prof.Profile
+            _, _, profile = load_distributed(pathlib.Path(f'profiles/{args.file}/profile.json'))
+            cat_href = profile.imports[0].href
+            cat_name = fs.model_name_from_href_str(cat_href)
+
+            _, _, catalog = load_distributed(pathlib.Path(f'catalogs/{cat_name}/catalog.json'))
+
+            try:
+                logging.debug(f'Loading yaml header file {args.yaml_header}')
+                yaml_header = yaml.load(pathlib.Path(args.yaml_header).open('r'), yaml.FullLoader)
+            except yaml.YAMLError as e:
+                logging.warning(f'YAML error loading yaml header for ssp generation: {e}')
+                return 1
+            markdown_path = trestle_root / args.output
+
+            sections = None
+            if args.sections is not None:
+                section_tuples = args.sections.strip("'").split(',')
+                sections = {}
+                for section in section_tuples:
+                    if ':' in section:
+                        s = section.split(':')
+                        sections[s[0]] = s[1]
+                    else:
+                        sections[section] = section
+
+            return self.generate_ssp(catalog, profile, markdown_path, sections, yaml_header)
+        else:
+            return self.assemble_ssp(args.file, args.output)
+
+    def _replace_params(self, text: str, control: cat.Control, param_dict: Dict[str, prof.SetParameter]) -> str:
+        # replace params with assignments from the profile
+        if control.params is not None:
+            for param in control.params:
+                set_param = param_dict.get(param.id, None)
+                if set_param is not None:
+                    text = text.replace(param.id, str(set_param.values[0].__root__))
+
+        # strip {{ }}
+        text = text.replace(' {{', '').replace(' }}', '').replace('insert: param, ', '').strip()
```

Comment:
See: https://github.com/IBM/compliance-trestle/issues/564 I believe this may be productive in the future. .. or we might just not consider it worthwhile (non-blocking).